src/views/Login/Trezor.js

```diff
@@ -1,146 +1,118 @@
 import * as bip32 from 'bip32';
-import React from 'react';
+import React, { useState } from 'react';
 import Maybe from 'folktale/maybe';
-import { Button } from '../../components/old/Base';
-import { Row, Col, H1, P } from '../../components/old/Base';
+import TrezorConnect from 'trezor-connect';
+import * as secp256k1 from 'secp256k1';
+
+import View from '../../components/View';
+import { ForwardButton } from '../../components/Buttons';
 import {
+  H1,
+  P,
   InnerLabel,
   Input,
   InnerLabelDropdown,
 } from '../../components/old/Base';
-import TrezorConnect from 'trezor-connect';
-import * as secp256k1 from 'secp256k1';
 
 import { TREZOR_PATH } from '../../lib/trezor';
-import { compose } from '../../lib/lib';
-import { withWallet } from '../../store/wallet';
-
-class Trezor extends React.Component {
-  constructor(props) {
-    super(props);
+import { useWallet } from '../../store/wallet';
 
-    this.state = {
-      hdpath: TREZOR_PATH.replace(/x/g, 0),
-      account: 0,
-    };
+export default function Trezor({ loginCompleted }) {
+  const { wallet, setWallet, setWalletHdPath } = useWallet();
 
-    this.handleAccountSelection = this.handleAccountSelection.bind(this);
-    this.handleHdPathInput = this.handleHdPathInput.bind(this);
-    this.pollDevice = this.pollDevice.bind(this);
-  }
+  const [hdPath, setHdPath] = useState(TREZOR_PATH.replace(/x/g, 0));
+  const [account, setAccount] = useState(0);
 
-  handleAccountSelection(account) {
-    let hdpath = this.state.hdpath;
+  const handleAccountSelection = account => {
+    setAccount(account);
     if (account !== 'custom') {
-      hdpath = TREZOR_PATH.replace(/x/g, account);
+      setHdPath(TREZOR_PATH.replace(/x/g, account));
     }
-    this.setState({ account, hdpath });
-  }
-
-  handleHdPathInput(hdpath) {
-    this.setState({ hdpath });
-  }
-
-  async pollDevice() {
-    const { setWallet, setWalletHdPath } = this.props;
-    const { hdpath } = this.state;
+  };
 
+  const pollDevice = async () => {
     TrezorConnect.manifest({
       email: 'bridge-trezor@urbit.org',
       appUrl: 'https://github.com/urbit/bridge',
     });
-    TrezorConnect.getPublicKey({ path: hdpath }).then(info => {
+    TrezorConnect.getPublicKey({ path: hdPath }).then(info => {
       if (info.success === true) {
         const payload = info.payload;
         const publicKey = Buffer.from(payload.publicKey, 'hex');
         const chainCode = Buffer.from(payload.chainCode, 'hex');
         const pub = secp256k1.publicKeyConvert(publicKey, true);
         const hd = bip32.fromPublicKey(pub, chainCode);
         setWallet(Maybe.Just(hd));
-        setWalletHdPath(hdpath);
+        setWalletHdPath(hdPath);
       } else {
         setWallet(Maybe.Nothing());
       }
     });
+  };
+
+  let accountOptions = [
+    {
+      title: 'Custom path',
+      value: 'custom',
+    },
+  ];
+  for (let i = 0; i < 20; i++) {
```

Comment:
looks like we can share this code with ledger a bit?

Reply:
Eh, the Trezor one is subtly different from Ledger, in that it also has a "custom path" option prepended. Makes this kind of awkward, since JS doesn't let you do `[head-item, rest-of-the-array]`, does it?

Reply:
it does! what about some code like

```js
const buildHardwarePathOptions = (head) => [head, ...times(20, i => ({ ... })]
```